src/Orchard.Web/Modules/Orchard.MessageBus/Services/DistributedShellStarter.cs

```diff
@@ -16,6 +17,10 @@ public class DistributedShellStarter : IDistributedShellStarter, IOrchardShellEv
 
         private readonly IMessageBus _messageBus;
 
+        private static readonly object locker = new object();
+
+        public static bool IsStarting { get; private set; }
```

Comment:
This is wrong, static values are the same across tenants. It should be some `IDependency` baked state.